packages/backend/src/registries/LocalRepositoryRegistry.ts

```diff
@@ -0,0 +1,59 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import type { LocalRepository } from '@shared/src/models/ILocalRepository';
+import { MSG_LOCAL_REPOSITORY_UPDATE } from '@shared/Messages';
+import { type Webview, Disposable } from '@podman-desktop/api';
+
+/**
+ * The LocalRepositoryRegistry is responsible for keeping track of the directories where recipe are cloned
+ */
+export class LocalRepositoryRegistry {
+  // Map path => LocalRepository
+  private repositories: Map<string, LocalRepository> = new Map();
+
+  constructor(private webview: Webview) {}
+
+  register(localRepository: LocalRepository): Disposable {
```

Comment:
This registry will be available the time you are in the same instance of the AI Studio app. But how will you build it again when you restart the app (either Podman Desktop or the extension?). Will you have the pair path/recipeId somewhere to be able to rebuild this "database"?

Reply:
Yes this is the goal _in the future_. For now we do not support adding manually folder/projects/repositories provided by the user. But we would at some point, and we want a place to store that information see https://github.com/projectatomic/ai-studio/issues/234

Today this is simply in memory, but yes in the future we would have to store that information somewhere on disk. But this is not part of this PR

Reply:
The solution for the models on disk is to use the modelId as directory name and to rely on it, so we can rebuild this database by scanning the directories in the dedicated directory. Could it be applied here also?

If we don't want to handle the manually created repositories for now, we will have only one repository on disk for each recipe, right? Even if I switch the model, if I understand correctly.

Reply:
> The solution for the models on disk is to use the modelId

I am not talking about the models id here, but folders containing `ai-studio.yaml` file

> If we don't want to handle the manually created repositories for now, we will have only one repository on disk for each recipe, right? Even if I switch the model, if I understand correctly.

Yes exactly, for now we will only have the cloned repository inside the LocalRepositoryRegistry

Reply:
> I am not talking about the models id here, but folders containing ai-studio.yaml file

Yes, I understand, but for me the problem of getting the path of the model's directory is the same as getting the recipy/repository's directory. The reason why I'm comparing both

Reply:
Yes sure